examples/op-errors/src/main.rs

```diff
@@ -49,10 +49,13 @@ where
         .iter()
         .map(|port| {
             ReceiveBatch::new(port.clone())
-                .map(|p| p.parse::<Ethernet>())
-                .filter(|p| p.ether_type() == EtherTypes::Ipv6)
+                .filter(|p| {
+                    let ethernet = p.peek::<Ethernet>().unwrap();
```

Comment:
maybe instead of unwrap, do a `if let ... else`.

Reply:
for sure. I was being to 1:1 there.